rpc/amqp-rpc.go

```diff
@@ -278,10 +287,45 @@ func (rpc *AmqpRPCServer) Start() (err error) {
 					Body:          jsonResponse, // XXX-JWS: jws.Sign(privKey, body)
 				})
 		}
+		finished <- true
 	}()
 	return
 }
 
+// HandleInterrupts creates a Goroutine to sit and watch for INT and TERM signals
+// and cancel the server consumer if it sees one so servers can gracefully shutdown.
+func (rpc *AmqpRPCServer) HandleInterrupts() (chan bool, error) {
+	stopWatching := make(chan bool, 1)
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGTERM)
+	signal.Notify(sigChan, os.Interrupt)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	consumerName := fmt.Sprintf("%s.%s", rpc.serverQueue, hostname)
```

Comment:
This duplicates the logic to create a consumerName above. Please pull it out into a helper function.